elements/lisk-p2p/src/p2p.ts

```diff
@@ -876,31 +883,55 @@ export class P2P extends EventEmitter {
 		const peerDiscoveryResponseLength = this._config.peerDiscoveryResponseLength
 			? this._config.peerDiscoveryResponseLength
 			: DEFAULT_MAX_PEER_DISCOVERY_RESPONSE_LENGTH;
+		const maxPeerInfoSize = this._peerPool.peerConfig.maxPeerInfoSize;
+		const wsMaxPayload = this._config.wsMaxPayload
+			? this._config.wsMaxPayload
+			: DEFAULT_WS_MAX_PAYLOAD;
+
+		const peerstWithSharedState = this._peerBook.allPeerstWithSharedState;
 
-		const knownPeers = this._peerBook.allPeers;
 		/* tslint:disable no-magic-numbers*/
 		const min = Math.ceil(
-			Math.min(peerDiscoveryResponseLength, knownPeers.length * 0.25),
+			Math.min(
+				peerDiscoveryResponseLength,
+				peerstWithSharedState.length * 0.25,
+			),
 		);
 		const max = Math.floor(
-			Math.min(peerDiscoveryResponseLength, knownPeers.length * 0.5),
+			Math.min(peerDiscoveryResponseLength, peerstWithSharedState.length * 0.5),
 		);
 		const random = Math.floor(Math.random() * (max - min + 1) + min);
 		const randomPeerCount = Math.max(
 			random,
-			Math.min(minimumPeerDiscoveryThreshold, knownPeers.length),
+			Math.min(minimumPeerDiscoveryThreshold, peerstWithSharedState.length),
 		);
 
-		const selectedPeers = shuffle(knownPeers)
-			.slice(0, randomPeerCount)
-			.map(
-				sanitizeOutgoingPeerInfo, // Sanitize the peerInfos before responding to a peer that understand old peerInfo.
-			);
+		const selectedPeers = shuffle(peerstWithSharedState).slice(
+			0,
+			randomPeerCount,
+		);
+
+		const validatedPeerList: ProtocolPeerInfo[] = [];
+
+		selectedPeers.forEach(peer => {
```

Comment:
in the other file, `for (const x of y)` is used, so prefer to keep it consistent